scripts/extensions/sams/src/store/sets/actions.ts

```diff
@@ -60,9 +60,19 @@ export function loadSets(): IThunkAction<Array<ISetItem>> {
     return (dispatch, _getState, {api}) => {
         return api.sets.getAll()
             .then((sets: Array<ISetItem>) => {
-                dispatch(receiveSets(sets));
-
-                return Promise.resolve(sets);
+                let set_ids: string[] = sets.map((set) => {
+                    return set._id
+                })
+                return api.assets.getCount(set_ids)
```

Comment:
And instead of enhancing the `IAssetItem` here you could call the new `ThunkAction` to load and store the set counts counts every time `loadSets` is called